srf_generation/Fault.py

```diff
@@ -344,7 +344,7 @@ def _calculate_dimensions(self):
         self._dbottom = (
             self._depth + np.sin(np.radians(self._dip)) * self.width / 2 + shift
         )
-        if self._dbottom > NHM_SEISMOGENIC_DEPTH:
+        if self.magnitude_scaling_relation == MagnitudeScalingRelations.LEONARD2014 and self._dbottom > NHM_SEISMOGENIC_DEPTH:
```

Comment:
This is only for shallow crustal rather than only doing it for leonard. In practice it's the same; but just keeping the conditional in the same logic that we made the decision is a good idea.